pkg/storegateway/indexheader/encoding/factory.go

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package encoding
+
+import (
+	"bufio"
+	"encoding/binary"
+	"hash/crc32"
+	"os"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const bufferSize = 4096
```

Comment:
This looks very small. Thoughts on this?

Reply:
I don't have any justification for this number. Since they're being pooled, I suppose it doesn't hurt to make them larger. My only concern would be if increasing the size of this buffer means `os.File.Read()` reads more data every time it's invoked.

Reply:
These are the questions we should answer running a benchmark on a real TSDB block.

Reply:
I would add a comment saying that this number was chosen without any justification, so future readers or maintainers would not hesitate to tune it if benchmarks show that it can be improved.